backend/src/api/events.js

```diff
@@ -0,0 +1,196 @@
+      response = 'Invalid Description'
+    } else {
+      const newEvent = new Event({
+        name: eventName,
+        date: eventDate,
+        startTime: eventStartTime,
+        endTime: eventEndTime,
+        location: eventLocation,
+        description: eventDescription,
+        attendees: []
+      })
+      await newEvent.save()
+      code = 200
+      success = true
+    }
+    res.json({
+      code,
+      message: response,
+      result: {},
+      success
+    })
+  })
+)
+
+// update an event
+router.put(
+  '/:event_id',
+  errorWrap(async (req, res) => {
+    const data = req.body
+    let response = 'Event Edited Sucessfully'
+    let eventId = req.params.event_id
+    let newName = data.name
+    let newDate = data.date
+    let newStartTime = data.startTime
+    let newEndTime = data.endTime
+    let newLocation = data.location
+    let newDescription = data.description
+
+    if (eventId == undefined) {
+      response = 'Invalid Edit Event Request'
+    } else {
+      if (newName != undefined) {
+        Event.findOneAndUpdate(
+          { _id: new mongodb.ObjectId(eventId) },
+          { $set: { name: newName } },
+          { new: true },
+          function(err, doc) {}
+        )
+      }
+      if (newDate != undefined) {
+        Event.findOneAndUpdate(
+          { _id: new mongodb.ObjectId(eventId) },
+          { $set: { date: newDate } },
+          { new: true },
+          function(err, doc) {}
+        )
+      }
+      if (newStartTime != undefined) {
+        Event.findOneAndUpdate(
+          { _id: new mongodb.ObjectId(eventId) },
+          { $set: { startTime: newStartTime } },
+          { new: true },
+          function(err, doc) {}
+        )
+      }
+      if (newEndTime != undefined) {
+        Event.findOneAndUpdate(
+          { _id: new mongodb.ObjectId(eventId) },
+          { $set: { endTime: newEndTime } },
+          { new: true },
+          function(err, doc) {}
+        )
+      }
+      if (newLocation != undefined) {
+        Event.findOneAndUpdate(
+          { _id: new mongodb.ObjectId(eventId) },
+          { $set: { location: newLocation } },
+          { new: true },
+          function(err, doc) {}
+        )
+      }
+      if (newDescription != undefined) {
+        Event.findOneAndUpdate(
+          { _id: new mongodb.ObjectId(eventId) },
+          { $set: { description: newDescription } },
+          { new: true },
+          function(err, doc) {}
+        )
+      }
+    }
+    res.json({
+      code: 200,
+      message: response,
+      result: {},
+      success: true
+    })
+  })
+)
+
+// delete an event
+router.delete(
+  '/:event_id',
+  errorWrap(async (req, res) => {
+    let response = 'Event Deleted Sucessfully'
+    let id = req.params.event_id
+    const event = await Event.findById(id)
+    if (event === undefined) {
+      response = 'Invalid Delete Event Request'
+    } else {
+      await Event.deleteOne({ _id: new mongodb.ObjectId(id) })
```

Comment:
I like all the error checking, but we're using two mongo requests here instead of just one. We can probably just use `deleteOne`. It'll either throw an error in which case we can use a try catch, or it will return an object with attribute `deletedCount` which will be 0. I'm not sure which one, but try doing one of those to reduce the number of queries.